src/generator/models.ts

```diff
@@ -61,7 +61,7 @@ export async function generateModels(session: Session<CodeModel>): Promise<strin
     text += '\t\treturn\n';
     text += '\t} else if azcore.IsNullValue(v) {\n';
     text += '\t\tm[k] = nil\n';
-    text += '\t} else if !reflect.ValueOf(v).IsNil() {\n';
+    text += '\t} else if !reflect.ValueOf(v).IsZero() {\n';
```

Comment:
This is the change you'll see in the codegen.

Reply:
@jhendrixMSFT , I don't think IsZero is a good way to be used for empty judgement here. If we set the param with false/0/"", this param will be ignored with unmarshall which may lose some useful info.

Reply:
Good catch.  This is because we're looking at the value, not the type.  I'll have an update later today.

Reply:
Fixed